airbyte-integrations/connectors/source-jira/source_jira/components/extractors.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from dataclasses import dataclass
+from typing import Any, List, Mapping, Union
+
+import dpath.util
+import requests
+from airbyte_cdk.sources.declarative.extractors import DpathExtractor
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from requests_cache import Response
+
+
+@dataclass
+class LabelsRecordExtractor(DpathExtractor):
+    """
+    A custom record extractor is needed to handle cases when records are represented as list of strings insted of dictionaries.
+    Example:
+        -> ["label 1", "label 2", ..., "label n"]
+        <- [{"label": "label 1"}, {"label": "label 2"}, ..., {"label": "label n"}]
+    """
+
+    def extract_records(self, response: Response) -> List[Mapping[str, Any]]:
+        records = super().extract_records(response)
+        return [{"label": record} for record in records]
+
+
+@dataclass
+class UnionListsRecordExtractor(DpathExtractor):
```

Comment:
Please add unit test to demonstrate why do we really need this component (or remove it):
Existed unit tests can be bypassed using standard DpathExtractor implementation:
![image](https://github.com/airbytehq/airbyte/assets/36314070/c747e866-fc72-44e1-a955-a358aa121cf8)
![image](https://github.com/airbytehq/airbyte/assets/36314070/15644250-bc6a-418b-b19a-ffb6ae53a5c0)


Nit: please use `pytest`'s `ids` to name tests instead of comments.

Reply:
Indeed there is no need in this custom component. This functionality is suported natively. Fixed